pkgs/by-name/pa/parabolic/package.nix

```diff
@@ -1,77 +1,94 @@
-{ lib
-, buildDotnetModule
-, fetchFromGitHub
-, dotnetCorePackages
-, gtk4
-, libadwaita
-, pkg-config
-, wrapGAppsHook4
-, glib
-, shared-mime-info
-, gdk-pixbuf
-, blueprint-compiler
-, python3
-, ffmpeg
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  gettext,
+  itstool,
+  ninja,
+  yelp-tools,
+  pkg-config,
+  libnick,
+  boost,
+  curl,
+  glib,
+  shared-mime-info,
+  libsecret,
+  gtk4,
+  libadwaita,
+  wrapGAppsHook4,
+  libxmlxx5,
+  blueprint-compiler,
+  yt-dlp,
+  ffmpeg,
+  aria2,
 }:
 
-buildDotnetModule rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "parabolic";
-  version = "2024.5.0";
+  version = "2024.11.1";
 
   src = fetchFromGitHub {
     owner = "NickvisionApps";
     repo = "Parabolic";
-    rev = version;
-    hash = "sha256-awbCn7W7RUSuEByXxLGrsmYjmxCrwywhhrMJq/iM1Uc=";
-    fetchSubmodules = true;
+    rev = "${finalAttrs.version}";
+    hash = "sha256-Vm5QVmPc0uipLlDy5Wn9S7HQ/0rLg0wVk0XdWRECQDY=";
   };
 
-  dotnet-sdk = dotnetCorePackages.sdk_8_0;
-  dotnet-runtime = dotnetCorePackages.runtime_8_0;
-  pythonEnv = python3.withPackages(ps: with ps; [ yt-dlp ]);
-
-  projectFile = "NickvisionTubeConverter.GNOME/NickvisionTubeConverter.GNOME.csproj";
-  nugetDeps = ./deps.nix;
-  executables = "NickvisionTubeConverter.GNOME";
-
-   nativeBuildInputs = [
+  nativeBuildInputs = [
+    cmake
+    gettext
+    itstool
+    ninja
+    yelp-tools
     pkg-config
     wrapGAppsHook4
+    blueprint-compiler
     glib
     shared-mime-info
-    gdk-pixbuf
-    blueprint-compiler
   ];
 
-  buildInputs = [ gtk4 libadwaita ];
-
-  runtimeDeps = [
+  buildInputs = [
+    libnick
+    boost
+    curl
+    libsecret
```

Comment:
```suggestion
```
Neither of these are declared in the CMake files. They should be covered by `libnick`

Reply:
The package fails to compile if either curl or libsecret is absent from `buildInputs`.  Don't know how to resolve this.

Reply:
```
2 files changed, 5 insertions(+), 5 deletions(-)
pkgs/by-name/li/libnick/package.nix   | 8 +++++---
pkgs/by-name/pa/parabolic/package.nix | 2 --

modified   pkgs/by-name/li/libnick/package.nix
@@ -45,15 +45,17 @@ stdenv.mkDerivation (finalAttrs: {
   buildInputs =
     [
       boost
-      curl
       maddy-markdown
     ]
     ++ lib.optionals stdenv.hostPlatform.isUnix [
       glib
       openssl
     ]
-    ++ lib.optional stdenv.hostPlatform.isWindows sqlcipher
-    ++ lib.optional withLibsecret libsecret;
+    ++ lib.optional stdenv.hostPlatform.isWindows sqlcipher;
+
+  propagatedBuildInputs = [
+    curl
+  ] ++ lib.optional withLibsecret libsecret;
 
   cmakeFlags = [
     (lib.cmakeBool "BUILD_TESTING" finalAttrs.finalPackage.doCheck)
modified   pkgs/by-name/pa/parabolic/package.nix
@@ -52,8 +52,6 @@ stdenv.mkDerivation (finalAttrs: {
   buildInputs = [
     libnick
     boost
-    curl
-    libsecret
     glib
     gtk4
     libadwaita
```
Would using propagatedBuildInputs be a correct solution here?

Reply:
This should work, yes